shell/platform/darwin/ios/framework/Source/FlutterPlugins.mm

```diff
@@ -0,0 +1,41 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/darwin/ios/framework/Headers/FlutterPlugins.h"
+
+@implementation FlutterMethodPlugin
+- (instancetype)initWithController:(FlutterViewController *)controller
+                       channelName:(NSString*)channelName {
+  NSAssert(channelName, @"Channel name cannot be nil");
+  return [self initWithController:controller
+                          channel:[FlutterMethodChannel methodChannelWithName:channelName
+                                                              binaryMessenger:controller]];
+}
+
+- (instancetype)initWithController:(FlutterViewController *)controller
+                           channel:(FlutterMethodChannel*)channel {
+  NSAssert(controller, @"Controller cannot be nil");
+  NSAssert(channel, @"Channel cannot be nil");
+  self = [super init];
+  NSAssert(self, @"Super init cannot be nil");
+  _controller = controller;
+  _channel = channel;
+  __unsafe_unretained id weakSelf = self;
```

Comment:
I am not sure why you used `__unsafe_unretained` here when `__block` would have worked fine. If we had enabled ARC, you would have had to use `__weak`. In any case, you are getting rid of the call handler on the `dealloc`, so it does not matter too much I suppose. Something like `__block typeof(self) weakSelf = self;` seems should work a lot better though. This will warn if the method signature `handleInvocationOfMethod:arguments:result:` changed because you are no longer calling a selector on an `id`.